mono/metadata/class.c

```diff
@@ -289,31 +289,48 @@ mono_type_name_check_byref (MonoType *type, GString *str)
 		g_string_append_c (str, '&');
 }
 
+static char*
+escape_special_chars (const char* identifier)
+{
+	// Assume the worst case, and thus only allocate once
+	char *res = g_malloc (strlen (identifier) * 2 + 1);
+	char *res_ptr = res;
+	for (const char *s = identifier; *s != 0; s++) {
+		switch (*s) {
+		case ',':
+		case '+':
+		case '&':
+		case '*':
```

Comment:
Consider a bitset with these characters. Or some way to avoid duplicating them. A macro. Etc.